elasticdl/python/worker/worker.py

```diff
@@ -304,39 +304,30 @@ def _update_local_model(self):
     def get_model(self):
         self._timing.start_record_time("get_model")
         if self._distribution_strategy != DistributionStrategy.ALLREDUCE:
-            variable_future_and_id_pairs = []
-            for ps_id, stub in enumerate(self._ps_stubs):
-                if ps_id not in self._ps_client.ps_to_parameter:
-                    continue
-                # async grpc call
-                req = elasticdl_pb2.PullDenseParametersRequest()
-                req.version = self._model_versions_from_ps[ps_id]
-                var_future = stub.pull_dense_parameters.future(req)
-                variable_future_and_id_pairs.append((var_future, ps_id))
-
-            for var_future, ps_id in variable_future_and_id_pairs:
-                res = var_future.result()
-                if not res.initialized:
-                    # push variable to ps for initialization
-                    parameters = [
-                        Tensor(name, self._non_embed_vars[name].numpy(), None)
-                        for name in self._ps_client.ps_to_parameter[ps_id]
-                    ]
-                    self._ps_client.push_dense_parameters(
-                        parameters, ps_id, self._model_versions_from_ps[ps_id]
+            dense_params, uninit_ps = self._ps_client.pull_dense_parameters(
+                [i for i in range(self._ps_num)], self._model_versions_from_ps
+            )
+
+            for ps_id in uninit_ps:
+                # push variable to ps for initialization
+                parameters = [
+                    Tensor(name, self._non_embed_vars[name].numpy(), None)
+                    for name in self._ps_client.ps_to_parameter[ps_id]
+                ]
+                self._ps_client.push_dense_parameters(
+                    parameters, ps_id, self._model_versions_from_ps[ps_id]
+                )
+                dense_params, uninit = self._ps_client.pull_dense_parameters(
```

Comment:
`dense_params` will be overridden by the last PS in uninit_ps and params from other PSs are lost here.

Reply:
Done. Fix it.